std/algorithm/iteration.d

```diff
@@ -4835,13 +4837,64 @@ if (isInputRange!Range && is(typeof(binaryFun!pred(r.front, r.front)) == bool))
     assert(equal(uniq([ 1, 1, 2, 1, 1, 3, 1]), [1, 2, 1, 3, 1]));
 }
 
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+
+    struct S
+    {
+        int i;
+        string s;
+    }
+
+    auto arr = [ S(1, "a"), S(1, "b"), S(2, "c"), S(2, "d") ];
+
+    // Let's consider just the 'i' member for equality
+    auto r = arr.uniq!((a, b) => a.i == b.i);
+    assert(r.equal([S(1, "a"), S(2, "c")]));
+    assert(r.front == S(1, "a"));
+    assert(r.back == S(2, "c"));
+
+    auto r2 = arr.uniq!((a, b) => a.i == b.i)(PickingPolicy.last);
+    assert(r2.equal([S(1, "b"), S(2, "d")]));
+    assert(r2.front == S(1, "b"));
+    assert(r2.back == S(2, "d"));
+
+    auto arr2 = [ S(1, "a"), S(1, "b") ];
+    auto r3 = arr2.uniq!((a, b) => a.i == b.i);
+    assert(r3.front == S(1, "a"));
+    assert(r3.front == r3.back);
+}
+
+/**
+   Dictates how `uniq` elements should be picked. By default stop at
```

Comment:
I don't think the default policy should be a part of this comment; it should be in the function comments of uniq. (Or not even there as perhaps the documentation makes it clear what the default is? :) )